usaspending_api/submissions/management/commands/rm_submission.py

```diff
@@ -67,33 +29,19 @@ def signal_handler(signal, frame):
         signal.signal(signal.SIGINT, signal_handler)
         signal.signal(signal.SIGTERM, signal_handler)
 
-        # This will throw an exception and exit the command if the id doesn't exist
+        broker_submission_id = options["submission_id"][0]
+
         try:
-            submission = SubmissionAttributes.objects.get(broker_submission_id=options["submission_id"][0])
+            submission = SubmissionAttributes.objects.get(broker_submission_id=broker_submission_id)
         except ObjectDoesNotExist:
-            raise "Submission with broker id " + str(options["submission_id"][0]) + " does not exist"
-
-        self.logger.info('Getting childless submissions')
-        potentially_childless = self.instances_potentially_left_childless(submission)
-        print('done')
-
-        deleted = defaultdict(int)
-        deleted_total = 0
-        deletion_tally = submission.delete()
-        deleted_total += deletion_tally[0]
-        deleted.update(deletion_tally[1])
+            raise "Broker submission id {} does not exist".format(broker_submission_id)
```

Comment:
It is preferable to throw a specific exception, perhaps `raise KeyError("Broker submission id {} does not exist".format(broker_submission_id))`

[Lengthly SO answer](https://stackoverflow.com/a/24065533)